python_modules/libraries/dagster-dbt/dagster_dbt/components/dbt_project/component.py

```diff
@@ -240,37 +255,62 @@ def get_asset_check_spec(
         return self._base_translator.get_asset_check_spec(asset_spec, manifest, unique_id, project)
 
     @cached_property
-    def cli_resource(self):
-        return DbtCliResource(self.project)
+    def dbt_project(self) -> DbtProject:
+        if isinstance(self.project, DbtProject):
+            return self.project
+        else:
+            # TODO: some amount of hacking could be done to make this work, but we
+            # should probably think of a more systemic solution
+            raise DagsterInvariantViolationError("Unsupported for RemoteDbtProject")
 
     def get_asset_selection(
         self, select: str, exclude: str = DBT_DEFAULT_EXCLUDE
     ) -> DbtManifestAssetSelection:
         return DbtManifestAssetSelection.build(
-            manifest=self.project.manifest_path,
+            manifest=self.dbt_project.manifest_path,
             dagster_dbt_translator=self.translator,
             select=select,
             exclude=exclude,
         )
 
     def write_state_to_path(self, state_path: Path) -> None:
+        state_dir = state_path.parent
+        project_dir = state_dir / "project"
+        if isinstance(self.project, RemoteDbtProject):
+            self.project.fetch(project_dir)
+
+        dbt_project = self._resolve_dbt_project(project_dir)
+
         # compile the manifest
-        self.project.preparer.prepare(self.project)
-        # move the manifest to the correct path
-        shutil.copyfile(self.project.manifest_path, state_path)
+        dbt_project.preparer.prepare(dbt_project)
+        # copy the manifest file to the state path
+        shutil.copyfile(dbt_project.manifest_path, state_path)
+
+    def _resolve_dbt_project(self, state_path: Optional[Path]) -> DbtProject:
```

Comment:
Nit: could we rename this function for `_resolve_dbt_project_from_state_path` or something similar? For a moment, I got confused because there's also the resolution fn `resolve_dbt_project` in this script.